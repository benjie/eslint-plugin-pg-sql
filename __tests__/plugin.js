const CLIEngine = require("eslint").CLIEngine;
const plugin = require("..");

function sanitise(results) {
  return results.map(result => ({
    ...result,
    filePath: "[snip]",
  }));
}

let simpleFunction = `\
create function my_test_function(foo text, bar text) returns text as $$
var a = "42";
return A; // Undefined!
$$ language plv8 stable;
`;

let cli;
beforeAll(() => {
  cli = new CLIEngine({
    envs: [],
    extensions: ["sql"],
    ignore: false,
    rules: {
      "eol-last": 2,
      "no-console": 2,
      "no-undef": 2,
      quotes: 2,
      "spaced-comment": 2,
    },
    useEslintrc: false,
  });
  cli.addPlugin("markdown", plugin);
});

test("should run on .sql files", () => {
  const report = cli.executeOnText(simpleFunction, "test.sql");

  expect(report.results).toHaveLength(1);
  expect(report.results[0].messages[0].message).toEqual(`'A' is not defined.`);
  expect(sanitise(report.results)).toMatchSnapshot();
});
