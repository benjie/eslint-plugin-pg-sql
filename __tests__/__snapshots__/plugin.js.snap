// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should run on .sql files 1`] = `
Array [
  Object {
    "errorCount": 1,
    "filePath": "[snip]",
    "fixableErrorCount": 0,
    "fixableWarningCount": 0,
    "messages": Array [
      Object {
        "column": 8,
        "endColumn": 9,
        "endLine": 4,
        "line": 4,
        "message": "'A' is not defined.",
        "nodeType": "Identifier",
        "ruleId": "no-undef",
        "severity": 2,
        "source": "return A; // Undefined!",
      },
    ],
    "source": "create function my_test_function(foo text, bar text) returns text as $$
var a = \\"42\\";
return A; // Undefined!
$$ language plv8 stable;
",
    "warningCount": 0,
  },
]
`;
